receiver/httpcheckreceiver/scraper.go

```diff
@@ -37,6 +38,11 @@ func (h *httpcheckScraper) start(ctx context.Context, host component.Host) (err
 	var expandedTargets []*targetConfig
 
 	for _, target := range h.cfg.Targets {
+		if target.Timeout == 0 {
+			// Set a reasonable timeout to prevent hanging requests
```

Comment:
worthy of a breaking change changelog entry. Please create a changelog entry with `make chlog-entry`